src/main/java/io/stargate/sgv2/jsonapi/service/operation/ReadAttempt.java

```diff
@@ -166,4 +180,60 @@ static class ReadResult {
       this.pagingState = CqlPagingState.from(resultSet);
     }
   }
+
+  static class ReadAttemptRetryPolicy<T extends TableBasedSchemaObject> extends RetryPolicy {
+
+    private RetryContext<T> retryContext = null;
+
+    ReadAttemptRetryPolicy() {
+      super(1, Duration.ofMillis(1));
+    }
+
+    void setRetryContext(RetryContext<T> retryContext) {
+      this.retryContext = retryContext;
+    }
+
+    @Override
+    public boolean shouldRetry(Throwable throwable) {
+
+      if (retryContext == null) {
+        throw new IllegalStateException("retryContext must not be null");
+      }
+      // clear the retry context so that we don't keep a reference to the last statement
+      var currentRetryContext = retryContext;
+      retryContext = null;
+
+      var allowFilteringMissing =
+          throwable instanceof InvalidQueryException
+              && throwable.getMessage().endsWith("use ALLOW FILTERING");
+
+      if (allowFilteringMissing) {
+        if (LOGGER.isWarnEnabled()) {
```

Comment:
I think we definitely want this log line, correct?
I could image we may need to dig in some patterns or get some 'complains' from ALLOW FILTERING.

So, do we want to just log it anyway.
I don't know we set LOGGER.isWarnEnabled() anywhere.

Reply:
the next highest level to log is ERROR and dont think it is an error, happy to keep as an warn for now. Added a ticket for metrics for retry and retry when allow filtering https://github.com/stargate/data-api/issues/1583